examples/lock-app/efr32/src/LockManager.cpp

```diff
@@ -407,6 +445,165 @@ bool LockManager::SetCredential(chip::EndpointId endpointId, uint16_t credential
     return true;
 }
 
+DlStatus LockManager::GetWeekdaySchedule(chip::EndpointId endpointId, uint8_t weekdayIndex, uint16_t userIndex,
+                                         EmberAfPluginDoorLockWeekDaySchedule & schedule)
+{
+    if ((weekdayIndex < MIN_SCHEDULE_INDEX) || (weekdayIndex > MAX_SCHEDULE_PER_USER))
+    {
+        ChipLogError(Zcl, "Cannot get schedule - schedule index is out of range [endpoint=%d,index=%d]", endpointId, weekdayIndex);
+        return DlStatus::kFailure;
+    }
+    if ((userIndex < MIN_USER_INDEX) || (userIndex > DOOR_LOCK_MAX_USERS))
```

Comment:
This check is repeated in many places. Cold we centralize it? Like seeing:

```cpp
   VerifyOrReturnValue(IsValidUserIndex(userIndex), DlStatus::kFailure);
```

where `bool IsValidUserIndex` is defined somewhere and is quite trivial and can log if the user index does not look valid. Or if we log endpoint we could have `IsValidUserIndexOnEndpoint` or similar.

Reply:
I have a question on this one. Should I check for a valid user index in the application code? From what I see, door-lock-server.cpp is supposed to check for valid user and credentials indices. There were some earlier bugs that allowed invalid indices to slip through from the server to my application when testing, hence why I added these checks.

Just worried about duplicating code.

Reply:
I believe it is worth checking - treat all input data as potentially wrong either due to bugs or using this in unexpected cases (e.g. lock-server.cpp users this, but in the future others may add more code paths with different logic).

The one-indexing is very odd ... I wish that was removed with the cluster code. Am explicitly not asking to change that in this PR though as that feels very yak-shaving-like.

Reply:
Added range checking helper function to each set/get user, credential and schedule function.